zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/identity/RoleAddEntityProcessor.java

```diff
@@ -92,7 +93,7 @@ final var record = command.getValue();
 
     final var entityId = record.getEntityId();
     final var entityType = record.getEntityType();
-    if (!isEntityPresent(entityId, entityType)) {
+    if (!allowNonExistingEntity(command, entityType) && !isEntityPresent(entityId, entityType)) {
```

Comment:
🔧 With this change we end up testing the user/client type twice, I'd suggest that one change would be to just pass the `command` into the method and then pull all the information needed from it there